datasets/ehealth_kd/ehealth_kd.py

```diff
@@ -0,0 +1,159 @@
+_TEST_DIR = "testing/scenario1-main/"
+_TEXT_FILE = "scenario.txt"
+_ANNOTATIONS_FILE = "scenario.ann"
+
+
+class eHealthKD(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="ehealth_kd", version=VERSION, description="eHealth Knowledge Discovery dataset"),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "sentence": datasets.Value("string"),
+                    "entities": [
+                        {
+                            "ent_id": datasets.Value("string"),
+                            "ent_text": datasets.Value("string"),
+                            "ent_tag": datasets.Value("string"),
+                            "start_character": datasets.Value("int32"),
+                            "end_character": datasets.Value("int32"),
+                        }
+                    ],
+                    "relations": [
+                        {
+                            "rel_id": datasets.Value("string"),
+                            "rel_tag": datasets.Value("string"),
+                            "arg1": datasets.Value("string"),
+                            "arg2": datasets.Value("string"),
+                        }
+                    ],
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls_to_download = {
+            "train": [f"{_URL}{_TRAIN_DIR}{_TEXT_FILE}", f"{_URL}{_TRAIN_DIR}{_ANNOTATIONS_FILE}"],
+            "dev": [f"{_URL}{_DEV_DIR}{_TEXT_FILE}", f"{_URL}{_DEV_DIR}{_ANNOTATIONS_FILE}"],
+            "test": [f"{_URL}{_TEST_DIR}{_TEXT_FILE}", f"{_URL}{_TEST_DIR}{_ANNOTATIONS_FILE}"],
+        }
+
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"text_dir": downloaded_files["train"][0], "ann_dir": downloaded_files["train"][1]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"text_dir": downloaded_files["dev"][0], "ann_dir": downloaded_files["dev"][1]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"text_dir": downloaded_files["test"][0], "ann_dir": downloaded_files["test"][1]},
+            ),
+        ]
+
+    def _generate_examples(self, text_dir, ann_dir):
+        """ Yields examples. """
+        with open(text_dir, encoding="utf-8") as t, open(ann_dir, encoding="utf-8") as a:
```

Comment:
It would be nice to rename `t` as e.g. `txt_file` and `a` as `ann_file` so the the code downstream is a bit easier to read

Reply:
Nice suggestion, done!